tests/tree_xml_parser_paragraph_processor_tests.py

```diff
@@ -127,3 +137,21 @@ def test_separate_intro_negative(self):
         intro, rest = _ExampleProcessor().separate_intro(nodes)
         self.assertIsNone(intro)
         self.assertEqual(nodes, rest)
+
+        # A node containing only a table:
+        xml_table = etree.fromstring('<GPOTABLE>stuff</GPOTABLE>')
+        table_node = Node(label=[mtypes.MARKERLESS], source_xml=xml_table)
+        nodes = [table_node]
+        intro, rest = _ExampleProcessor().separate_intro(nodes)
+        self.assertIsNone(intro)
+        self.assertEqual(nodes, rest)
+
+        # A table node and another node:
+        nodes = [table_node, Node(label=['a'])]
+        intro, rest = _ExampleProcessor().separate_intro(nodes)
+        self.assertIsNone(intro)
+        self.assertEqual(nodes, rest)
+        nodes = [table_node, Node(label=[mtypes.MARKERLESS])]
+        intro, rest = _ExampleProcessor().separate_intro(nodes)
+        self.assertIsNone(intro)
+        self.assertEqual(nodes, rest)
```

Comment:
Similar to the above; would it make sense to split these into separate tests and maybe combine some of the repetition?